src/hooks/index.js

```diff
@@ -123,6 +124,15 @@ function activate(agent) {
 
   logger = agent.logger;
 
+  // Plugin stuff
+  var api = new Plugin(agent);
+  Object.keys(agent.plugins).forEach(function(pluginName) {
+    if (toInstrument[pluginName]) {
```

Comment:
With this check, a plugin will only be loaded if its name is already listed in `toInstrument`, but new plugins will most likely be for modules not listed in `toInstrument`.